cmd/clusterctl/client/repository/components.go

```diff
@@ -431,9 +431,11 @@ func fixRBAC(objs []unstructured.Unstructured, targetNamespace string) ([]unstru
 			// assign a namespaced name
 			b.Name = fmt.Sprintf("%s-%s", targetNamespace, b.Name)
```

Comment:
@yastij pointed out last week that the name also gets changes, which I see here it'd show up with the `targetNamespace` which for the webhook case wouldn't be correct?

Reply:
I have checked with @yastij this is acceptable.

Reply:
Seems a little off to rename something that's not being installed in our namespace 🤷

Reply:
If we can come out with a simple rule to identify ClusterRoleBindings to not rename I can implement it; using ClusterRoleBindings that has a subject in the `capi-web-hook` namespace is a little bit fragile for me...

However I'm fine either way, just pick your preferred option
@yastij ^^

Reply:
If we assign the name after checking subjects, we should be able to determine if it's in the webhook namespace or not — if it is, we can change the targetNamespace to the capi-webhook one within this case condition

Reply:
@vincepri - I think we should rename the CRB and CR, you'd expect the list of verb/resources different per webhook, otherwise we might endup in a situation where the last applied `infrastructure-components.yaml` wins